### PR TITLE
Add X-Scope-Probe-ID header to POSTs

### PR DESCRIPTION
**Pull request by peterbourgon**
/cc @2opremio @paulbellamy

### CODE REVIEW

**Comment by paulbellamy** [comment]
Would something like the hostname be more meaningful to the user?

**Comment by peterbourgon** [comment]
Hmm, I can't think of a normal scenario where the hostname wouldn't work... will make the change.

File: xfer/publisher.go [hunk 1]
```diff
@@ -22,10 +22,18 @@ type Publisher interface {
 type HTTPPublisher struct {
 	url   string
 	token string
+	id    string
 }
 
+// ScopeProbeIDHeader is the header we use to carry the probe's unique ID. The
+// ID is generated at runtime and has the same lifetime as the probe process.
```
Comment (paulbellamy) [inline xfer/publisher.go]: This isn't true anymore.

**Comment by paulbellamy** [comment]
Fix comment, please, then LGTM

**Comment by peterbourgon** [comment]
Test failure seems to be GCE related.

**Pull request merged**